tools/saqc/V2/gen.py

```diff
@@ -0,0 +1,1124 @@
+def get_methods(module):
+    methods_with_saqc = []
+    classes = inspect.getmembers(module, inspect.isclass)
+    for name, cls in classes:
+        if inspect.ismodule(cls):
+            continue
+        methods = inspect.getmembers(cls, inspect.isfunction)
+        for method_name, method in methods:
+            try:
+                parameters = inspect.signature(method).parameters
+                if "self" in parameters:
+                    self_param = parameters["self"]
+                    annotation_str = None
+                    if isinstance(self_param.annotation, str):
+                        annotation_str = self_param.annotation.strip("'")
+                    elif isinstance(self_param.annotation, ForwardRef):
+                        annotation_str = self_param.annotation.__forward_arg__.strip(
+                            "'"
+                        )
+                    elif hasattr(self_param.annotation, "__name__"):
+                        annotation_str = self_param.annotation.__name__
+
+                    if annotation_str == "SaQC":
+                        methods_with_saqc.append(method)
+            except (ValueError, TypeError) as e:
+                sys.stderr.write(
+                    f"Warning: Could not inspect signature for {cls.__name__}.{method_name}: {e}\n"
+                )
+                continue
+    return methods_with_saqc
+
+
+def _split_type_string_safely(type_string: str) -> list[str]:
+    """
+    Zerlegt einen Typ-String bei '|' oder ',', ignoriert aber Trennzeichen
+    innerhalb von Klammern (eckig, rund).
+    """
+    parts = []
+    current_part = ""
+    bracket_level = 0
+    for char in type_string:
+        if char in ('[', '('):
+            bracket_level += 1
+        elif char in (']', ')'):
+            bracket_level -= 1
+
+        if char in ('|', ',') and bracket_level == 0:
+            if current_part.strip():
+                parts.append(current_part.strip())
+            current_part = ""
+        else:
+            current_part += char
+
+    if current_part.strip():
+        parts.append(current_part.strip())
+
+    return [p for p in parts if p]
+
+
+def _create_param_from_type_str(type_str: str, param_name: str, param_constructor_args: dict, is_optional: bool) -> Optional[object]:
+    param_object = None
+    base_type_str = type_str.strip()
+    is_tuple = False
+
+    tuple_match = re.fullmatch(r"tuple\[\s*([^,]+).*", base_type_str, re.IGNORECASE)
+    if tuple_match:
+        is_tuple = True
+        inner_type_str = tuple_match.group(1).strip()
+        base_type_str = inner_type_str
+        param_constructor_args["label"] = param_constructor_args.get("label", inner_type_str) + " (one or more)"
+
+    creation_args = param_constructor_args.copy()
+
+    if is_tuple:
+        creation_args['multiple'] = True
+
+    if base_type_str in ('SaQCFields', 'NewSaQCFields'):
+        param_object = TextParam(argument=param_name, multiple=True, **creation_args)
```

Comment:
Ideally field would be a data_column parameter: https://docs.galaxyproject.org/en/latest/dev/schema.html#data-column